### PR TITLE
Add dynamic LH sidebar navigation. Refactor titles. Add i18n to sidebar

### PR DESCRIPTION
**Pull request by PBillingsby**
Don't merge. Work in progress.

### CODE REVIEW

**Comment by PBillingsby** [comment]
LINKED: https://github.com/Developer-DAO/DAO-job-board/issues/157

File: client/components/navigation/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
+import NextLink from 'next/link';
```
Comment (angeljgomezc) [inline client/components/navigation/Navbar.tsx:1]: this file is duplicated and in the wrong folder/location

**Review by angeljgomezc — CHANGES_REQUESTED** [review]

File: packages/client/components/navigation/Dashboard.tsx [hunk 2]
```diff
@@ -14,15 +14,19 @@ import {
...
 import Navbar from './Navbar';
-
-import { faClipboardList, faUsers } from '@fortawesome/free-solid-svg-icons';
+import HireSidebar from './HireSidebar';
+import EarnSidebar from './EarnSidebar';
 
 export default function Dashboard() {
   const sidebar = useDisclosure();
-
+  let isHiring = true;
```
Comment (angeljgomezc) [inline packages/client/components/navigation/Dashboard.tsx:23]: Can we navigate to the `earning` section with this? Can you add a temporary button or am I missing something? Looks good in the preview by the way, just would love to see what the `Earning` section looks like

File: packages/client/components/navigation/Dashboard.tsx [hunk 3]
```diff
@@ -65,90 +70,12 @@ export default function Dashboard() {
...
-              <Text size="md">
-                <FontAwesomeIcon icon={faClipboardList} /> Jobs List
-              </Text>
-            </Link>
-          </NextLink>
-        </Stack>
-
-        <Stack alignSelf="flex-end"></Stack>
-      </Stack>
+      {isHiring ? <HireSidebar /> : <EarnSidebar />}
```
Comment (frankTurtle) [inline packages/client/components/navigation/Dashboard.tsx:78]: Love seeing shit being cleaned up like this 💪

File: packages/client/components/navigation/NavLink.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import NextLink from 'next/link';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+
+interface LinkProps {
+  path: string;
+  icon: IconProp;
+  title: string;
+}
+[];
```
Comment (frankTurtle) [inline packages/client/components/navigation/NavLink.tsx:11]: What's this lil guy for?

File: packages/client/components/navigation/NavTitle.tsx [hunk 5]
```diff
@@ -12,15 +12,13 @@ export default function NavTitle() {
...
-    } else if (router.pathname === '/post-job') {
-      return 'Post a Job';
-    } else if (router.pathname === '/create-profile') {
-      return 'Create a Profile';
-    }
+    const dynamicTitles: any = {
+      '/developers': 'Developers For Hire',
+      '/jobs': 'Job List',
+      '/post-job': 'Post a Job',
+      '/create-profile': 'Create a Profile',
```
Comment (frankTurtle) [inline packages/client/components/navigation/NavTitle.tsx:19]: These should have translations, no?

**Review by frankTurtle — COMMENTED** [review]
My only comments would be ( aside from the ones I left ) 

* If you want some feedback I tend to add the `[WIP]` tag to the title and you can set this to be a Draft PR ( I'm not sure how you created the PR, but just thought I'd toss that out there )

image

File: packages/client/components/navigation/Dashboard.tsx (hunk 2 above)
Comment (PBillingsby) [reply packages/client/components/navigation/Dashboard.tsx:23]: This is temporary. I can mock a toggle button tomorrow connecting these conditions. I’ll send a screenshot on discord

File: packages/client/components/navigation/NavLink.tsx (hunk 4 above)
Comment (PBillingsby) [reply packages/client/components/navigation/NavLink.tsx:11]: Type checking the incoming props from the Hire sidebar like so

```
<NavLink
           path="/my_projects"
           icon={faClipboardList}
           title={t('components.hire_sidebar.my_projects')}
         />
   ```

File: packages/client/components/navigation/NavTitle.tsx (hunk 5 above)
Comment (PBillingsby) [reply packages/client/components/navigation/NavTitle.tsx:19]: @frankTurtle you’re  absolutely right. I’m going to go through and fix that up when these are all ready to be reviewed again

File: packages/client/components/navigation/NavLink.tsx (hunk 4 above)
Comment (PBillingsby) [reply packages/client/components/navigation/NavLink.tsx:11]: Or the Earn Sidebar

Comment (frankTurtle) [reply packages/client/components/navigation/NavLink.tsx:11]: I'm meaning the `[];` That's part of it?

Comment (PBillingsby) [reply packages/client/components/navigation/NavLink.tsx:11]: I was previously trying to map the Navlink props from an array of objects and I’m not well versed with ts so just forgot to remove them when I explicitly called the component

File: client/components/navigation/Navbar.tsx (hunk 1 above)
Comment (PBillingsby) [reply client/components/navigation/Navbar.tsx:1]: Where is the duplicate to this?

Comment (angeljgomezc) [reply client/components/navigation/Navbar.tsx:1]: The file, not the line haha

**Comment by PBillingsby** [comment]
Closing due to recent decisions to use stricter UI choices for components, spacing, typography...

**Comment by angeljgomezc** [comment]
Hey @PBillingsby! Can you finish this so we can merge?

**Pull request closed without merging**